lib/ld-eventsource/impl/event_parser.rb

```diff
@@ -37,12 +37,16 @@ def items
               event = maybe_create_event
               reset_buffers
               gen.yield event if !event.nil?
-            else
-              case line
-                when /^(\w+): ?(.*)$/
-                  item = process_field($1, $2)
-                  gen.yield item if !item.nil?
-              end
+            elsif (colon = line.index(':'))
+              name = line.slice!(0...colon)
+              next unless name.match?(/^\w+$/)
+
+              # delete the colon followed by an optional space
+              line.delete_prefix!(':')
+              line.delete_prefix!(' ')
```

Comment:
I can definitely see how this change will greatly reduce the memory usage, particularly when the payload isn't a comment.

However, if a comment is sent, I think the next bit of logic can have its memory consumption further reduced.

Mutating the string in place seems to take more memory than calling the non-destructive `delete_prefix`. Maybe something like this?

```suggestion
              name = line.slice(0...colon)
              next unless name.match?(/^\w+$/)

              # delete the colon followed by an optional space
              line = line.slice(colon...).delete_prefix(':').delete_prefix(" ")
```

What do you think?

Reply:
> Mutating the string in place seems to take more memory than calling the non-destructive delete_prefix

I accept that that's a finding you've seen, but it's very counter-intuitive to me in terms of how allocating a brand new string could possibly take _less_ memory than mutating a string in place, so I'm puzzled.

Reply:
And if we are OK with allocating a brand-new string, then I think it's safe to say that the most efficient approach would be to find out how many characters we want to exclude and then just call `substring` with an offset that is past those characters, rather than making these three separate calls to compute intermediate strings.

Reply:
That's really interesting that mutating in-place takes more memory, i would never have guessed that. I'm ok with this change, the regex was the big culprit, the strings were more minor

Reply:
I would still prefer to be doing a single `slice!` here, rather than `slice!` followed by `delete_prefix!` twice. If `line` contains a very large piece of data (which is the use case we're concerned about in the first place) then those two `delete_prefix!` calls are each causing the whole buffer to be shifted left one character at a time. It should be simple enough to instead just add 1 to `colon` (there is no real decision being made by that first `delete_prefix`, we already know for sure that the character at that position is `':'`), then check if the next character (if any) is a space and increment it again if so.

Reply:
I mean, I understand that the regex usage was a bigger problem and this is better than that, but that's no reason to still be incurring unnecessary overhead in the improved code.

Reply:
What I'm saying is that you can remove both of them. There is no need to mutate the string more than once, and I'd prefer that we not do so, given that the point of these changes is to reduce unnecessary overhead when the string is very long.

```ruby
name = line.slice(0...colon)
colon += 1
colon += 1 if colon < line.length && line[colon] == ' '
line.slice!(colon...)
```